rewrite-java/src/main/java/org/openrewrite/java/ChangePackage.java

```diff
@@ -113,14 +116,45 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                     }
                     stopAfterPreVisit();
                 }
+                if (tree instanceof SourceFileWithReferences) {
+                    SourceFileWithReferences cu = (SourceFileWithReferences) tree;
+                    boolean recursive = Boolean.TRUE.equals(ChangePackage.this.recursive);
+                    String recursivePackageNamePrefix = oldPackageName + ".";
+                    for (Reference ref : cu.getReferences().getReferences()) {
+                        if (ref.getKind().equals(Reference.Kind.PACKAGE) && ref.getValue().equals(oldPackageName) || recursive && ref.getValue().startsWith(recursivePackageNamePrefix)) {
```

Comment:
Did you mean? Without these parentheses it looks like we might match Kind.TYPE against a recursive prefix.
```suggestion
                        if (ref.getKind().equals(Reference.Kind.PACKAGE) && 
                           (ref.getValue().equals(oldPackageName) || recursive && ref.getValue().startsWith(recursivePackageNamePrefix))) {
```

Reply:
I made a mistake in my train of thought, as `Type` also contains a package, which should be changed here, so I'm removing the `ref.getKind().equals(Reference.Kind.PACKAGE)` statement I added